### ISSUE TITLE
[import/export] False "Multiple exports of name" error when two "export * as ..." contain same name. 

### ISSUE THREAD

**Issue by zimmed** [benmosher/eslint-plugin-import#1834]
*Apologies if this is a duplicate -- it's difficult to search terms with an asterisk.*

Found issue using:
  `eslint` 6.6.0, 6.8.0, 7.3.1
  `eslint-plugin-import` 2.20.2, 2.21.2
(tried various versions to see if it had already been fixed)

## Setup:
```javascript
// A.js
export const FOO = 'a-foobar';
```
```javascript
// B.js
export const FOO = 'b-foobar';
```
```javascript
// C.js
export * as A from './A';
export * as B from './B';
```

## Expected:
No `import/export` error in `C.js`, since both instances of `FOO` are under their own namespaces (A/B).
```javascript
import * as C from './C';

console.log(C.A.FOO); // 'a-foobar'
console.log(C.B.FOO); // 'b-foobar'
console.log(C.FOO); // undefined
```

## Result:
Error in C.js:
`Multiple exports of name 'FOO'. eslint(import/export)`

## Workaround

The following equivalent method works without errors, but annoying because ESLint is preventing me from taking advantage of newer ES syntax:

```javascript
import * as A from './A';
import * as B from './B';

export { A, B };
```

**Comment by ljharb** [comment]
This might be a duplicate of #518?

**Comment by zimmed** [comment]
> This might be a duplicate of #518?

That seems to be related to shadowing, whereas this example has no actual variable shadowing. If it was...
```javascript
export * from './A';
export * from './B';
```
Then that would likely be the issue, but since the original code posted is equivalent to:
```javascript
import * as A from './A';
import * as B from './B';

export { A, B };
```
There are no shadowed variable names, since `FOO` only exists within the individual namespaces. (This works without lint errors, by the way).
